### PR TITLE
Provide access to content of framebuffer

### PR DESCRIPTION
**Pull request by derfred**
I would like to access the content of the framebuffer programatically. I have added two helper functions.

Adresses #209

### CODE REVIEW

File: core/display.js [hunk 1]
```diff
@@ -224,6 +224,14 @@ export default class Display {
         this.viewportChangePos(0, 0);
     }
 
+    copyImageData() {
+        return this._targetCtx.getImageData(0, 0, this.width, this.height)
```
Comment (CendioOssman) [inline core/display.js:228]: Why `_targetCtx`? It seems more reasonable to get the uncropped, full display.

File: core/display.js [hunk 2]
```diff
@@ -224,6 +224,14 @@ export default class Display {
...
+    }
+
+    toDataURL(format) {
```
Comment (CendioOssman) [inline core/display.js:231]: Let's mimic the API fully and also include `encoderOptions`.

On the same theme, let's keep the same argument names as the canvas API and rename this to `type`.

Comment (CendioOssman) [inline core/display.js:231]: The modern way is to use blobs, so a `toBlob()` would be nice as well.

**Review by CendioOssman — CHANGES_REQUESTED** [review]
Seems reasonable. We try to mimic existing JavaScript APIs as much as possible though, so there as some tweaks I'd like.

Also, I'd like `docs/API.md` to be updated with these new methods.

File: core/display.js (hunk 1 above)
Comment (derfred) [reply core/display.js:228]: I've switched the calls to use the `_drawCtx` and the `_backbuffer`

**Comment by derfred** [comment]
@CendioOssman I've addressed your comments. Let me know what you think

**Review by CendioOssman — CHANGES_REQUESTED** [review]
Looks great! Thanks for addressing all the comments!

Please squash these to a single commit and I can go ahead and merge things.

Also, try to add a little bit more context to the commit message. The summary in this PR is a very good example.

**Comment by derfred** [comment]
@CendioOssman I've squashed the commits

**Review by CendioOssman — APPROVED** [review]

**Pull request merged**